src/common/harvest/harvest-scheduler.js

```diff
@@ -76,8 +85,8 @@ export class HarvestScheduler extends SharedContext {
         }
       }
     } else {
-      const runAfterSending = opts?.unload ? undefined : onHarvestFinished // don't bother running onFinish handler if this is the final harvest
-      this.harvest.sendX(this.endpoint, opts, runAfterSending)
+      // const runAfterSending = opts?.unload ? undefined : onHarvestFinished // don't bother running onFinish handler if this is the final harvest
+      this.harvest.sendX(this.endpoint, opts, onHarvestFinished)
```

Comment:
why does it matter if the page thinks about the runAfterSending part?

If the page actually offloaded, it will never get to the point that it can call onHarvestFinished since it uses sendBeacon

`if (cbFinished && method === submitData.xhr) { ...` 

If it didnt "really" offload, it will call the onHarvestFinished() cb since it used xhr.

Reply:
Need to remove lingering commented out code